src/components/map/Map.tsx

```diff
@@ -157,131 +144,264 @@ export const Map = ({
     };
   }
 
+  useEffect(() => {
+    if (clusteringEnabled) {
+      const index = new Supercluster({
+        radius: clusterDistance,
+        maxZoom: 20
+      });
+
+      const points = locations.map((location, index) => ({
+        type: 'Feature',
+        properties: { cluster: false, id: location.id, index },
+        geometry: {
+          type: 'Point',
+          coordinates: [location.position.longitude, location.position.latitude]
+        }
+      }));
+
+      index.load(points);
+      const bounds = [-180, -85, 180, 85];
+      setClusters(index.getClusters(bounds, zoomLevel));
+    } else {
+      setClusters(
+        locations.map((location, index) => ({
+          type: 'Feature',
+          properties: { cluster: false, id: location.id, index },
+          geometry: {
+            type: 'Point',
+            coordinates: [location.position.longitude, location.position.latitude]
+          }
+        }))
+      );
+    }
+
+    // Berechne die Begrenzungskoordinaten für alle Marker
+    if (locations.length > 0 && isInitialFit) {
+      const coordinates = locations.map((loc) => [loc.position.longitude, loc.position.latitude]);
+      if (locations.length === 1) {
+        if (cameraRef.current) {
+          cameraRef.current.setCamera({
+            centerCoordinate: coordinates[0],
+            zoomLevel: 15,
+            animationDuration: 0
+          });
+          setTimeout(() => {
+            setIsLoading(false);
+          }, 4000); // Timeout in Millisekunden anpassen je nach Bedarf
+          setIsInitialFit(false); // Verhindert zukünftige automatische Anpassungen
+        }
+      } else {
+        const [minLng, minLat] = coordinates.reduce(
+          (prev, curr) => [Math.min(prev[0], curr[0]), Math.min(prev[1], curr[1])],
+          [Infinity, Infinity]
+        );
+        const [maxLng, maxLat] = coordinates.reduce(
+          (prev, curr) => [Math.max(prev[0], curr[0]), Math.max(prev[1], curr[1])],
+          [-Infinity, -Infinity]
+        );
+        const deltaLng = (maxLng - minLng) * 0.1; // Add 10% padding to the bounds
+        const deltaLat = (maxLat - minLat) * 0.1; // Add 10% padding to the bounds
```

Comment:
please move this heavy maths in a separate function outside of the component with a speaking name and some comment.
this will lead to more readability inside the component.

Reply:
thank you, will be done in the upcoming commit.